### PR TITLE
Commit "minimal" and "latest" lockfiles

### PR DESCRIPTION
**Pull request by apoelstra**
This is a proposed strategy for maintaining tested lockfiles in the rust-bitcoin ecosystem. The idea is that we would have both a "minimal" and a "latest" lockfile, and in both cases we have trusted crev reviews for all dependencies (this is not implemented here, I'd like to start by committing the lockfiles so we can agree what to review).

Periodically we can update the "latest" one to reflect new versions of deps that we've gotten around to reviewing.

I have local nix build scripts that are able to test every commit of proposed PRs against both lockfiles. In CI it is probably reasonable to at least do `cargo test --locked --all-features` with both of them, to make sure that the tests at least pass on each PR with them.

Thoughts?

### CODE REVIEW

**Review by tcharding — DISMISSED** [review]
ACK 1b12cc5f5854efdd14153b0937bb4ae8a2c43abf

**Comment by tcharding** [comment]
The idea is then that we would each publish crev reviews for all the crates (and versions) in both lock files, right?

File: tests/serde.rs
Comment (Kixunil) [inline tests/serde.rs:86]: It's really weird that different engines encode it differently but I don't care much; it's not our fault.

**Review by Kixunil — APPROVED** [review]
ACK 1b12cc5f5854efdd14153b0937bb4ae8a2c43abf

Didn't verify the lock files.

**Comment by apoelstra** [comment]
Great, thanks both of you!

We need a rust-secp maintainer to ACK before I can merge but it sounds like we're agreed on strategy here.

**Comment by apoelstra** [comment]
@elichai @sanket1729 @jonasnick could one of you take a look and ACK this?

**Review by sanket1729 — APPROVED** [review]
ACK bd9d3c9de7bc1d98b3d82fc4434ca997bc705ce7. Verified the lockfiles. The latest one has a couple lines diff, but that is expected :) . minimal one is the same

**Pull request merged**
